app/lib/service/email/backend.dart

```diff
@@ -62,84 +60,81 @@ class EmailBackend {
       if (stopAfter != null && sw.elapsed > stopAfter) break;
       if (m.isNotAlive) continue;
       if (!m.mayAttemptNow) continue;
-      final wasSent = await _trySendOutgoingEmail(m.uuid);
-      if (wasSent) {
-        successful++;
-      }
+      final count = await _trySendOutgoingEmail(m.uuid);
+      successful += count;
     }
     return successful;
   }
 
-  /// Tries to send [emails]. The [OutgoingEmail] entry will be deleted after
+  /// Tries to send [email]. The [OutgoingEmail] entry will be deleted after
   /// the email was sent successfully.
   ///
   /// This method should be called right after the entries
   /// are saved in the Datastore.
-  Future<void> trySendOutgoingEmails(Iterable<OutgoingEmail> emails) async {
-    for (final id in emails.map((e) => e.uuid)) {
-      await _trySendOutgoingEmail(id);
-    }
+  Future<void> trySendOutgoingEmail(OutgoingEmail email) async {
+    await _trySendOutgoingEmail(email.uuid);
   }
 
   /// Tries to send email with the given [id]. The
   /// [OutgoingEmail] entry will be deleted after the
   /// email was sent successfully.
   ///
-  /// Returns true if the email was sent successfully.
-  Future<bool> _trySendOutgoingEmail(String id) async {
+  /// Returns the number of emails that were sent successfully.
+  Future<int> _trySendOutgoingEmail(String id) async {
     final key = _db.emptyKey.append(OutgoingEmail, id: id);
     final now = clock.now().toUtc();
+    final claimId = createUuid();
     final entry = await withRetryTransaction(_db, (tx) async {
       final o = await tx.lookupOrNull<OutgoingEmail>(key);
-      if (o == null || o.isNotAlive) {
+      if (o == null || o.isNotAlive || o.claimId != null) {
         return null;
       }
       o.attempts++;
       o.lastAttempted = now;
       // retry after a random delay in the next 2-6 hours
```

Comment:
```suggestion
      // retry after a random delay in the next 2-6 hours, if and only if,
      // o.claimId has been cleared. We never retry sending emails if we don't know
      // if the email was sent or not (because we don't want to send it multiple times)
```